InfinispanAPIObject.adoc

```diff
@@ -0,0 +1,36 @@
+== Infinispan API
+
+=== Motivation
+
+Being able to offer a simple API to create clustered caches and improve API usability
```

Comment:
I think this type of code belongs to the user space. I wouldn't place it in Infinispan (but on a modules that interfaces Infinispan to Vert.x for example).